Documentation/installation/k8s-install-migration.rst

```diff
@@ -0,0 +1,310 @@
+        bpf:
+          hostLegacyRouting: true # Migration: Allow for routing between Cilium and the existing overlay
+
+4.  Optional: auto-detect settings using the :ref:`cilium-cli <install_cilium_cli>`.
+    This will consume the template and auto-detect any other relevant Helm values.
+    Review these values for your particular installation.
+
+    .. code-block:: shell-session
+
+        $ cilium install --helm-values values-migration.yaml --helm-auto-gen-values values-initial.yaml
+        $ cat values-initial.yaml
+
+5.  Install cilium using :ref:`helm <k8s_install_helm>`.
+
+    .. code-block:: shell-session
+
+      $ helm repo add cilium https://helm.cilium.io/
+      $ helm install cilium cilium/cilium --namespace kube-system --values values-initial.yaml
+
+
+    At this point, you should have a cluster with Cilium installed and an overlay established, but no
+    pods managed by Cilium itself. You can verify this with the ``cilium`` command.
+
+    .. code-block:: shell-session
+
+      $ cilium status --wait
+      ...
+      Cluster Pods:     0/3 managed by Cilium
+
+
+6.  Create a :ref:`per-node config<per-node-configuration>` that will instruct Cilium to "take over" CNI networking
+    on the node. Initially, this will apply to no nodes; you will roll it out gradually via
+    the migration process.
+
+    .. code-block:: shell-session
+
+        cat <<EOF | kubectl apply --server-side -f -
+        apiVersion: cilium.io/v2alpha1
+        kind: CiliumNodeConfig
+        metadata:
+          namespace: kube-system
+          name: cilium-default
+        spec:
+          nodeSelector:
+            matchLabels:
+              io.cilium.migration/cilium-default: "true"
+          defaults:
+            write-cni-conf-when-ready: /host/etc/cni/net.d/05-cilium.conflist
+            custom-cni-conf: "false"
+            cni-chaining-mode: "none"
+            cni-exclusive: "true"
+        EOF
+
+Migration
+---------
+
+At this point, you are ready to begin the migration process. The basic flow is:
+
+Select a node to be migrated. It is not recommended to start with a control-plane node.
+
+.. code-block:: shell-session
+
+  export NODE="kind-worker" # for the kind example
+
+1.  **Optional: cordon and drain the node in question.** Depending on your individual requirements,
+    you may wish to cordon and drain before applying the migration.
+
+    **Pros:**
+
+    - Users are isolated from any potential issues
+    - Affected pods are moved to a new node and immediately restarted, rather than waiting for the node to reboot
+
+    **Cons:**
+
+    - Pods may be restarted twice during the migration
+    - Any potential issues may be noticed sooner
+
+    It is recommended to always cordon and drain at the beginning of the migration process, so that
+    end-users are not impacted by any potential issues.
+
+    .. code-block:: shell-session
+
+      $ kubectl cordon $NODE
+      $ kubectl drain $NODE
```

Comment:
When going through the tutorial, when I drain, I get some error messages draining the node from the Flannel and Cilium Daemonset.

Should we be draining the Node using `--ignore-daemonsets` ?

There are pending nodes to be drained:
 kind-worker
cannot delete DaemonSet-managed Pods (use --ignore-daemonsets to ignore): kube-flannel/kube-flannel-ds-rf9b4, kube-system/cilium-2j2lb, kube-system/install-cni-plugins-md4gh, kube-system/kube-proxy-zhxcg

Reply:
Yup, good catch.